### ISSUE TITLE
oce failing on macOS 10.12

### ISSUE THREAD

**Issue by xzcvczx** [Homebrew/homebrew-science#4330]
### Indicate troubleshooting steps taken

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Checked that a similar issue hasn't already been filed?
- [x] Looked up the software bug tracker if reporting a build error?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports / build errors:
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:81:5: error: redefinition of enumerator '_CLOCK_REALTIME'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:82:5: error: redefinition of enumerator '_CLOCK_MONOTONIC'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:83:5: error: redefinition of enumerator '_CLOCK_PROCESS_CPUTIME_ID'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:84:5: error: redefinition of enumerator '_CLOCK_THREAD_CPUTIME_ID'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:85:3: error: typedef redefinition with different types ('enum clockid_t' vs 'enum clockid_t')

#### Brief summary of issue:
build issues with oce on macOS 10.12 w/ xcode 8
https://github.com/tpaviot/oce/issues/643

#### The complete command that you issued:
brew install homebrew/science/oce

#### Gist with output of `brew doctor`:
Your system is ready to brew.

#### Link output by `brew gist-logs <formula>`:
https://gist.github.com/14ac5609e449a10eae22d7412748d3bd

#### Output of `brew config`:
HOMEBREW_VERSION: 1.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: d5b6ecfc5078041ddf5f61b259c57f81d5c50fcc
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 193811f15d210d2509e29e8664b5a164d7d15fb6
Core tap last commit: 83 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.8.4 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: N/A
macOS: 10.12-x86_64
Xcode: 8.0
CLT: N/A
X11: N/A

**Comment by ilovezfs** [comment]
This should fix it for you :)
```
diff --git a/oce.rb b/oce.rb
index 8226ea7..1196b75 100644
--- a/oce.rb
+++ b/oce.rb
@@ -18,9 +18,11 @@ class Oce < Formula
   depends_on "freeimage" => :recommended
   depends_on "gl2ps" => :recommended
   depends_on "tbb" => :recommended
-  depends_on :macos => :snow_leopard
+  depends_on macos: :snow_leopard
 
-  conflicts_with "opencascade", :because => "OCE is a fork for patches/improvements/experiments over OpenCascade"
+  conflicts_with "opencascade", because: "OCE is a fork for patches/improvements/experiments over OpenCascade"
+
+  patch :DATA if MacOS.version >= :sierra
 
   def install
     cmake_args = std_cmake_args
@@ -57,3 +59,31 @@ class Oce < Formula
     assert_equal "1", shell_output(cmd).chomp
   end
 end
+
+__END__
+diff --git a/adm/cmake/TKernel/CMakeLists.txt b/adm/cmake/TKernel/CMakeLists.txt
+index 54e241e..0c293f1 100644
+--- a/adm/cmake/TKernel/CMakeLists.txt
++++ b/adm/cmake/TKernel/CMakeLists.txt
+@@ -29,7 +29,7 @@ if(WIN32)
+ 	set(TOOLKIT_LIBS ${TOOLKIT_LIBS} ${CSF_SOCKETLibs_LIB} ${CSF_advapi32_LIB} ${CSF_gdi32_LIB} ${CSF_user32_LIB} ${CSF_kernel32_LIB} ${CSF_psapi_LIB})
+ else(WIN32)
+ 	#  An implementation for Mac OS X has been added in src/OSD/gettime_osx.h
+-	if(NOT APPLE)
++	if(APPLE)
+ 		include( CheckFunctionExists )
+ 		check_function_exists( clock_gettime CLOCK_GETTIME_IN_LIBC )
+ 		if(NOT CLOCK_GETTIME_IN_LIBC)
+diff --git a/src/OSD/OSD_Chronometer.cxx b/src/OSD/OSD_Chronometer.cxx
+index 444f844..e3b986c 100644
+--- a/src/OSD/OSD_Chronometer.cxx
++++ b/src/OSD/OSD_Chronometer.cxx
+@@ -51,7 +51,7 @@
+   #include <mach/mach.h>
+ #endif
+ 
+-#if defined(__APPLE__) && defined(__MACH__)
++#if defined(__OPPLE__) && defined(__MACH__)
+ #include "gettime_osx.h"
+ #endif
+ 
```

**Comment by xzcvczx** [comment]
Thanks, i just hacked around it with a #ifndef __MAC_10_12 in gettime_osx.h

**Comment by ilovezfs** [comment]
yup that's what the `__OPPLE__` is effectively

**Comment by dpo** [comment]
Fixed in #4406.
